docs/REST API/rest-api.md

```diff
@@ -6932,12 +6938,104 @@ Deletes the session specified by ID. When the user associated with the session n
+| per_page                | integer | query | Results per page.                                                                                                                                                          |
+| order_key               | string  | query | What to order results by. Allowed fields are `name`, `hosts_count`, `vulnerabilities_count`. Default is `hosts_count` (descending).      |
+| order_direction         | string  | query | **Requires `order_key`**. The direction of the order given the order key. Options include `asc` and `desc`. Default is `asc`.                                              |
+| query                   | string  | query | Search query keywords. Searchable fields include `title` and `cve`.                                                                                             |
+| team_id                 | integer | query | _Available in Fleet Premium_ Filters the software to only include the software installed on the hosts that are assigned to the specified team.                             |
+| vulnerable              | bool    | query | If true or 1, only list software that has detected vulnerabilities. Default is `false`.                                                                                    |
+
+#### Example
+
+`GET /api/v1/fleet/software/titles`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "counts_updated_at": "2022-01-01 12:32:00",
+  "count": 2,
+  "software_titles": [
+    {
+      "id": 12,
+      "name": "Firefox.app",
+      "versions_count": 3,
+      "source": "apps",
+      "hosts_count": 48,
+      "versions": [ 
+        {
+          "id": 123,
+          "version": "1.12",
+          "vulnerabilities": ["CVE-2023-1234","CVE-2023-4321","CVE-2023-7654"]
+        }, 
+        {
+          "id": 124, 
+          "version": "3.4",
+          "vulnerabilities": ["CVE-2023-1234","CVE-2023-4321","CVE-2023-7654"]
+        },
+        {
+          "id": 12
+          "version": "1.13",
+          "vulnerabilities": ["CVE-2023-1234","CVE-2023-4321","CVE-2023-7654"]
+        }
+      ]  
+    },
+    {
+      "id": 22,
+      "name": "Google Chrome.app",
+      "versions_count": 5,
+      "source": "apps",
+      "hosts_count": "345",
+      "versions": [
+        {
+          "id": 331,
+          "version": "118.1",
+          "vulnerabilities": ["CVE-2023-1234"]
+        },
+        {
+          "id": 332,
+          "version": "119.0",
+          "vulnerabilities": ["CVE-2023-9876", "CVE-2023-2367"]
+        },
+        {
+          "id": 334,
+          "version": "119.4",
+          "vulnerabilities": ["CVE-2023-1133", "CVE-2023-2224"]
+        },
+        {
+          "id": 348,
+          "version": "121.5",
+          "vulnerabilities": ["CVE-2023-0987", "CVE-2023-5673", "CVE-2023-1334"]
```

Comment:
@noahtalerman @marko-lisica any thoughts on having `host_counts` here? so it's consistent with the `versions` in the `GET /software/titles/:id` endpoint

Reply:
I'll move forward with this as-is for now.

Reply:
The reason why `host_count` is not included here is because it's not used in UI, while it's used in the software titles details page.